src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -248,17 +263,44 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @return true if a right curly should be alone on a line.
      */
     private static boolean shouldBeAloneOnLine(RightCurlyOption bracePolicy, Details details) {
+        final int tokenLambda = details.lcurly.getParent().getType();
         return bracePolicy == RightCurlyOption.ALONE
+                && tokenLambda != TokenTypes.LAMBDA
                 && !isAloneOnLine(details)
                 && !isEmptyBody(details.lcurly)
-                || bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
+                || shouldBeAloneInLambda(bracePolicy, details)
+                || shouldBeAloneOrSingleLine(bracePolicy, details);
+    }
```

Comment:
I do not like implementation of this method now.
It i become even more hard to read than before.
Please clearly split RightCurlyOption.ALONE and RightCurlyOption.ALONE_OR_SINGLELINE
so code should be like 
```
return (bracePolicy == RightCurlyOption.ALONE 
              && ............)
          || (bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE 
              && ............)
```
or even completely split this method into two for ALONE and ALONE_OR_SINGLELINE , with check for policy in `validate`.